### ISSUE TITLE
Adding symmetric encryption filter option (feature request)

### ISSUE THREAD

**Issue by ssolari** [PyTables/PyTables#584]
The introduction of basic file systems into the cloud (e.g. AWS EFS) is opening the door now to using HDF5-Pytables in a much more scalable and productionized manner.   However, the need for at rest encryption exists in a variety of applications (note EFS does not provide encryption at rest).   

Today, a variety of workarounds exist for encrypting data inside a pytables file, however they don't provide an elegant or simple solution to full encyption at rest.   Basic encryption would be an awesome feature. 

Adding arbitrary encryption options is likely not practical, however, selecting a single filter option providing AES-256 symmetric encryption, for example, where the user is responsible for providing the 32byte key seems reasonably implemented.   Since AES-256 is currently the best practice defacto standard, it should meet the vast majority of encryption at rest needs and could be well thought out and implemented simply and safely.    Obviously, in any filter pipeline, compression would be performed before encryption. 

The request is for an additional filter pipeline step providing AES-256 encryption, with an encryption flag taking a 32byte string in the Filters class.

Are the maintainers open to putting some limited straight forward encryption in the feature pipeline?  Or is any sort of encryption not going to happen?

**Comment by FrancescAlted** [comment]
Yes, why not?  If you would like to provide a PR, we should be happy to merge it.

**Comment by ssolari** [comment]
I'm not sure if my skills are up to it or not.  I would definitely need some direction.   I can operate well at the python level, but anything down in the C realm might be too much time considering my efficiency there.    Could you provide some high level direction on where the hooks/changes would likely be?   I can then take a look.
